### PR TITLE
ENG-2152 distinguish custom bundle installations

### PR DESCRIPTION
**Pull request by firegloves**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **firegloves**

If they are not already assigned, you can assign the PR to them by writing `/assign @firegloves` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr116**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr116**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr116**

File: src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleInstallService.java
Comment (w-caffiero-entando) [inline src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleInstallService.java:196]: @firegloves not sure about this, a non-json install plan is a strange condition that should lead to a http status 422.
Are you may be checking the format before this code?

File: src/test/java/org/entando/kubernetes/controller/mockmvc/InstallFlowTest.java
Comment (w-caffiero-entando) [inline src/test/java/org/entando/kubernetes/controller/mockmvc/InstallFlowTest.java:251]: @firegloves please use "is" for flags (instead of "get").
You are indeed doing it for the install plan

File: src/main/java/org/entando/kubernetes/controller/digitalexchange/job/model/InstallPlan.java
Comment (w-caffiero-entando) [inline src/main/java/org/entando/kubernetes/controller/digitalexchange/job/model/InstallPlan.java:146]: @firegloves very briefly explain what a custom installation is

File: src/main/java/org/entando/kubernetes/model/job/EntandoBundleJobEntity.java
Comment (w-caffiero-entando) [inline src/main/java/org/entando/kubernetes/model/job/EntandoBundleJobEntity.java:72]: @firegloves very briefly explain what is the purpose of the field

File: src/main/java/org/entando/kubernetes/service/digitalexchange/component/EntandoBundleServiceImpl.java
Comment (w-caffiero-entando) [inline src/main/java/org/entando/kubernetes/service/digitalexchange/component/EntandoBundleServiceImpl.java:109]: @firegloves I think populates essentially means "filling".
What about a simple `updateCustomInstallationFlag`?

File: src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleInstallService.java
Comment (firegloves) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleInstallService.java:196]: I added it because `objectMapper.writeValueAsString` declares to throw the `JsonProcessingException`. I could move it elsewhere but the serialization implies the exception management

File: src/main/java/org/entando/kubernetes/service/digitalexchange/component/EntandoBundleServiceImpl.java
Comment (firegloves) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/component/EntandoBundleServiceImpl.java:109]: it sounds good

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | e7d17c34c628037b9ff3654b068646a7f564e9b2 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 8d3c587cc6e51d11cf160634de21c77f2cba950f | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr116**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr116**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr116**

**Review by w-caffiero-entando — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando-k8s/entando-component-manager/pull/116#pullrequestreview-644698874" title="Approved">w-caffiero-entando</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)~~ [w-caffiero-entando]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 3ea569404edb22c1db6682f751c603c3968edb98 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr116**

**Pull request merged**
